app/models/release.rb

```diff
@@ -34,10 +34,7 @@ def author
     super || NullUser.new(author_id)
   end
 
-  def self.find_by_version(version)
-   if version =~ /\Av(\d+)\Z/
-     number = $1.to_i
-     find_by_number(number)
-   end
- end
+  def self.find_by_version!(version)
+    find_by_number!(version[/\Av(\d+)\Z/, 1].to_i)
```

Comment:
basically making sure it's always a NotFound / 404 instead of having to deal with custom logic, nobody should find with an unknown version so no need to add extra fancy logic